### PR TITLE
Trimming the run containers so that they use less memory and are safer.

### PR DESCRIPTION
**Pull request by lemire**
runContainer16.myOpts is subject to data races as indicated in PR https://github.com/RoaringBitmap/roaring/pull/299

It seems like this is an unnecessary component of the code that is better entirely removed. It is what the current PR does.

### CODE REVIEW

**Comment by lemire** [comment]
@fanyang01 Would you review my PR?

**Comment by lemire** [comment]
We seem to be doing a lot of memory size tests that are unnecessary.

**Comment by lemire** [comment]
These alloc. tests were pretty bad because they were platform dependent.

File: runcontainer_test.go [hunk 1]
```diff
@@ -366,81 +366,6 @@ func TestRleRunReverseIterator16(t *testing.T) {
 	})
 }
 
-func TestRleRunSearch16(t *testing.T) {
```
Comment (fanyang01) [inline runcontainer_test.go:369]: This test case should not be deleted.

**Review by fanyang01 — CHANGES_REQUESTED** [review]
`TestRleRunSearch16` should not be deleted?

File: runcontainer_test.go (hunk 1 above)
Comment (lemire) [reply runcontainer_test.go:369]: It relies on the old API. The function it tests has been deleted.

Comment (lemire) [reply runcontainer_test.go:369]: The test is described as '"RunContainer16.search should respect the prior bounds we provide for efficiency of searching through a subset of the intervals"'. We do not provide this functionality anymore, and it was never used in the actual engine.

**Comment by lemire** [comment]
@fanyang01 Thanks for the read-over. I will merge. We can add more tests later if we'd like.

**Pull request merged**